### PR TITLE
Feature-13480 Backend support for freeze and unfreeze filing

### PR DESCRIPTION
**Pull request by ritvick**
*Issue #:* /bcgov/entity/issues/13480

*Description of changes:*
Backend support for admin freeze filing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/filings/validations/admin_freeze.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+from typing import Dict, Optional
+
+from flask_babel import _ as babel  # noqa: N813, I004, I001; importing camelcase '_' as a name
+# noqa: I004
+from legal_api.errors import Error
+from legal_api.models import Business
+from ...utils import get_str  # noqa: I003; needed as the linter gets confused from the babel override above.
+
+
+def validate(business: Business, put_back_on: Dict) -> Optional[Error]:
```
Comment (argush3) [inline legal-api/src/legal_api/services/filings/validations/admin_freeze.py:25]: references to `put_back_on` in this function should be `admin_freeze` right?

Comment (ritvick) [reply legal-api/src/legal_api/services/filings/validations/admin_freeze.py]: correct, made the correction.

File: legal-api/src/legal_api/services/filings/validations/admin_freeze.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    """Validate the Court Order filing."""
+    if not business or not admin_freeze:
+        return Error(HTTPStatus.BAD_REQUEST, [{'error': babel('A valid business and filing are required.')}])
+    msg = []
+
+    if not get_str(admin_freeze, '/filing/adminFreeze/details'):
+        msg.append({'error': babel('Admin Freeze details are required.'), 'path': '/filing/adminFreeze/details'})
+
+    if not get_str(admin_freeze, '/filing/adminFreeze/freeze'):
+        msg.append({'error': babel('Admin Freeze flag is required.'), 'path': '/filing/adminFreeze/freeze'})
```
Comment (argush3) [inline legal-api/src/legal_api/services/filings/validations/admin_freeze.py:35]: I think we should add another check after this to ensure we are not trying to freeze/unfreeze a business that is already in the desired admin_freeze state.

File: legal-api/src/legal_api/services/filings/validations/admin_freeze.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    if business.admin_freeze == bool(get_str(admin_freeze, '/filing/adminFreeze/freeze')):
+        msg.append({'error': babel('Admin Freeze flag cannot be same as currunt state.'),
```
Comment (argush3) [inline legal-api/src/legal_api/services/filings/validations/admin_freeze.py:38]: typo for current

File: legal-api/src/legal_api/services/filings/validations/admin_freeze.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    """Validate the Court Order filing."""
+    if not business or not admin_freeze:
+        return Error(HTTPStatus.BAD_REQUEST, [{'error': babel('A valid business and filing are required.')}])
+    msg = []
+
+    if not get_str(admin_freeze, '/filing/adminFreeze/details'):
+        msg.append({'error': babel('Admin Freeze details are required.'), 'path': '/filing/adminFreeze/details'})
+
+    if not get_str(admin_freeze, '/filing/adminFreeze/freeze'):
+        msg.append({'error': babel('Admin Freeze flag is required.'), 'path': '/filing/adminFreeze/freeze'})
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/filings/validations/admin_freeze.py:35]: This is already validated in schema (as these are marked as required). So you can remove these from here

File: legal-api/src/legal_api/services/filings/validations/admin_freeze.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    current_state = business.admin_freeze or False
+
+    if current_state == bool(get_str(admin_freeze, '/filing/adminFreeze/freeze')):
+        msg.append({'error': babel('Admin Freeze flag cannot be same as current state.'),
+                    'path': '/filing/adminFreeze/freeze'})
+
+    if msg:
+        return Error(HTTPStatus.BAD_REQUEST, msg)
+    return None
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/filings/validations/admin_freeze.py:39]: Can you add a unit test for this file. You can refer existing testing file for validation

Comment (ritvick) [reply legal-api/src/legal_api/services/filings/validations/admin_freeze.py]: Added

File: legal-api/src/legal_api/services/filings/validations/admin_freeze.py (hunk 4 above)
Comment (ritvick) [reply legal-api/src/legal_api/services/filings/validations/admin_freeze.py]: removed

File: legal-api/src/legal_api/services/filings/validations/admin_freeze.py [hunk 6]
```diff
@@ -30,10 +40,11 @@ def validate(business: Business, admin_freeze: Dict) -> Optional[Error]:
 
     current_state = business.admin_freeze or False
 
-    if current_state == bool(get_str(admin_freeze, '/filing/adminFreeze/freeze')):
+    if current_state == str_to_bool(get_str(admin_freeze, '/filing/adminFreeze/freeze')):
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/filings/validations/admin_freeze.py:43]: use `get_bool` func which is available in utils instead of this

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by argush3 — APPROVED** [review]

**Pull request merged**